app/src/common/shared/com/igalia/wolvic/ui/adapters/ThumbnailAsyncTask.java

```diff
@@ -0,0 +1,143 @@
+package com.igalia.wolvic.ui.adapters;
+
+import android.content.ContentResolver;
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.media.ThumbnailUtils;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Build;
+import android.os.CancellationSignal;
+import android.os.Handler;
+import android.os.Looper;
+import android.provider.MediaStore;
+import android.util.Log;
+import android.util.Size;
+import android.webkit.MimeTypeMap;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+import com.igalia.wolvic.utils.SystemUtils;
+import com.igalia.wolvic.utils.UrlUtils;
+
+import java.io.File;
+import java.io.IOException;
+
+public class ThumbnailAsyncTask extends AsyncTask<Void, Void, Bitmap> {
+
+    static final String LOGTAG = SystemUtils.createLogtag(ThumbnailAsyncTask.class);
+
+    public interface OnSuccessDelegate {
+        void onSuccess(Bitmap bitmap);
+    }
+
+    private static final Size DEFAULT_SIZE = new Size(96, 96);
+    private static final int DEFAULT_SIZE_KIND = MediaStore.Images.Thumbnails.MICRO_KIND;
+
+    private final ContentResolver mContentResolver;
+    private final Uri mFileUri;
+    private final OnSuccessDelegate mOnSuccessDelegate;
+    private CancellationSignal mCancellationSignal;
+
+    public ThumbnailAsyncTask(@NonNull Context context, @NonNull Uri fileUri, OnSuccessDelegate onSuccessDelegate) {
+
+        Log.e(LOGTAG, "new ThumbnailAsyncTask");
```

Comment:
Nit. `Log.d`. Same for most (some are real errors) logging statements in this file.

Reply:
Fixed.